test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestExecutionContext.java

```diff
@@ -98,12 +98,11 @@ public ClientYamlTestResponse callApi(String apiName, Map<String, String> params
             }
         }
 
-        // Although include_type_name defaults to false, there is a large number of typed index creations
-        // in REST tests that need to be manually converted to typeless calls. As a temporary measure, we
-        // specify include_type_name=true in indices.create calls, unless the parameter has been set otherwise.
-        // This workaround will be removed once we convert all index creations to be typeless.
+        // When running tests against a mixed 7.x/6.x cluster we specify 'include_type_name=false' if it is not
+        // explicitly set. This allows us to omit the parameter in the test description, while still being able
+        // to communicate with 6.x nodes where include_type_name defaults to 'true'.
         if (apiName.equals("indices.create") && requestParams.containsKey(INCLUDE_TYPE_NAME_PARAMETER) == false) {
```

Comment:
Should we be checking the esVersion() here and only injecting this when we are on a mixed version cluster as in the checks below? I think it would be good to check 7.0 clusters with the default rather than injecting the parameter?

Reply:
Actually nevermind. I just saw in a later commit you are doing this check